### ISSUE TITLE
Support for .forward files

### ISSUE THREAD

**Issue by feld** [corecode/dma#37]
Noticed dma doesn't support .forward files and there's a placeholder in dma.c

**Comment by emaste** [comment]
@corecode and I have had some offline discussion about this. The suggestion is to add a small suid standalone helper program e.g. dma-read-forward that emits the desired forward entry on stdout.

Links to some commentary on .forward files:
- https://docstore.mik.ua/orelly/networking/sendmail/ch25_07.htm
- https://docs.oracle.com/cd/E19120-01/open.solaris/819-1634/mailrefer-39/index.html

**Comment by JohnHay** [comment]
Support for .forward files would be very useful. I think that is the only way for a normal user get their mail delivered somewhere else.

The lack of .forward support did cause a bit of pain recently when upgrading FreeBSD machines from 13.x to 14.0, which changed from using sendmail to dma by default. None of the machines needed to receive mail from outside, so that part did not concern me. But I did not expect dma to ignore .forward files.

**Comment by sstallion** [comment]
This is something that just bit me as well and is the only feature missing from dma to convince me to move over from Postfix. I'd be willing to take this on, but I'm not sure I have enough time to support much beyond a basic forward. Is there any interest in this?

**Comment by corecode** [comment]
the reason it's not implemented so far is that it seemed complicated to implement securely.  what if the home dir is 700?  how do we read .forward then?  what if it specifies a local mbox? how do we write to that file?  even more complicated to follow forwards around between local users.  that's why there is /etc/aliases support, which does not have any of these permission issues. 

if you can propose a simple i.e. by-inspection secure implementation, then great! 


On June 13, 2024 7:07:38 PM CDT, Steven Stallion ***@***.***> wrote:
>This is something that just bit me as well and is the only feature missing from dma to convince me to move over from Postfix. I'd be willing to take this on, but I'm not sure I have enough time to support much beyond a basic forward. Is there any interest in this?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/corecode/dma/issues/37#issuecomment-2166983423
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sstallion** [comment]
I agree, it's a quagmire. With respect to aliases, does dma support external addresses or must aliases always reference a local account?

**Comment by corecode** [comment]
you absolutely can specify a remote address.  it's really the reason for dma, to simply deliver remote. 


On June 13, 2024 7:52:09 PM CDT, Steven Stallion ***@***.***> wrote:
>I agree, it's a quagmire. With respect to aliases, does dma support external addresses or must aliases always reference a local account?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/corecode/dma/issues/37#issuecomment-2167020956
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sstallion** [comment]
I'll give that a shot. Thanks!

**Comment by JohnHay** [comment]
/etc/aliases works well for root (the administrator), but normal users
cannot edit it, so it is not a solution for them.
